File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/MinRule.java
package com.wesleyelliott.kubwa.rule;

/**
 * Created by wesley on 2016/07/28.
 */

public class MinRule extends Rule<Integer> {

    private int minValue;

    public MinRule(Integer minValue) {
        super(Integer.class);
        this.minValue = minValue;
    }

    @Override
    public boolean isValid(Integer value) {
        return value != null && value > minValue;
    }
}

File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/MinRule.kt
package com.wesleyelliott.kubwa.rule

/**
 * Created by wesley on 2016/07/28.
 */

class MinRule(private val minValue: Int) : Rule<Int>(Int::class.java) {

    override fun isValid(value: Int?): Boolean {
        return value != null && value > minValue
    }
}
